src/silx/gui/widgets/WaitingOverlay.py

```diff
@@ -1,109 +1,26 @@
 from silx.gui.widgets.WaitingPushButton import WaitingPushButton
 from silx.gui import qt
-from silx.gui.qt import inspect as qt_inspect
-from silx.gui.plot import PlotWidget
+from .OverlayMixIn import _OverlayMixIn
 
 
-class WaitingOverlay(qt.QWidget):
+class WaitingOverlay(_OverlayMixIn, WaitingPushButton):
     """Widget overlaying another widget with a processing wheel icon.
 
     :param parent: widget on top of which to display the "processing/waiting wheel"
     """
 
     def __init__(self, parent: qt.QWidget) -> None:
-        super().__init__(parent)
+        WaitingPushButton.__init__(self, parent)
+        _OverlayMixIn.__init__(self, parent)
         self.setContentsMargins(0, 0, 0, 0)
-
-        self._waitingButton = WaitingPushButton(self)
-        self._waitingButton.setDown(True)
-        self._waitingButton.setWaiting(True)
-        self._waitingButton.setStyleSheet(
+        self.setDown(True)
+        self.setWaiting(True)
+        self.setStyleSheet(
             "QPushButton { background-color: rgba(150, 150, 150, 40); border: 0px; border-radius: 10px; }"
         )
         self._registerParent(parent)
 
-    def text(self) -> str:
-        """Returns displayed text"""
-        return self._waitingButton.text()
-
     def setText(self, text: str):
         """Set displayed text"""
-        self._waitingButton.setText(text)
+        super().setText(text)
         self._resize()
```

Comment:
Is it needed to keep resize here?

Reply:
Yes, else we have display issues like:

![image](https://github.com/user-attachments/assets/2ecd52db-96d8-4b05-b127-51a3ef553fb6)

instead of 

![image](https://github.com/user-attachments/assets/6c5d4312-2757-4513-a0db-c6a11aaa74b1)